x-pack/plugins/infra/public/containers/logs/log_entries/index.ts

```diff
@@ -124,48 +144,99 @@ const useFetchEntriesEffect = (
   dispatch: Dispatch,
   props: LogEntriesProps
 ) => {
-  const { getLogEntriesAround, getLogEntriesBefore, getLogEntriesAfter } = useGraphQLQueries();
-
-  const [prevParams, cachePrevParams] = useState(props);
+  const [prevParams, cachePrevParams] = useState<LogEntriesProps | undefined>();
   const [startedStreaming, setStartedStreaming] = useState(false);
 
-  const runFetchNewEntriesRequest = async (override = {}) => {
+  const runFetchNewEntriesRequest = async (overrides: Partial<LogEntriesProps> = {}) => {
+    if (!props.startTimestamp || !props.endTimestamp) {
+      return;
+    }
+
     dispatch({ type: Action.FetchingNewEntries });
+
     try {
-      const payload = await getLogEntriesAround({
-        ...omit(props, 'jumpToTargetPosition'),
-        ...override,
-      });
+      const fetchArgs: LogEntriesRequest = {
+        sourceId: overrides.sourceId || props.sourceId,
+        startTimestamp: overrides.startTimestamp || props.startTimestamp,
+        endTimestamp: overrides.endTimestamp || props.endTimestamp,
+        query: overrides.filterQuery || props.filterQuery || undefined, // FIXME
+      };
+
+      if (props.timeKey) {
+        (fetchArgs as LogEntriesCenteredRequest).center = props.timeKey;
+      } else {
+        (fetchArgs as LogEntriesBeforeRequest).before = 'last';
+      }
```

Comment:
What do you think about the following formulation in order to avoid the explicit type casts?

```suggestion
      const commonFetchArgs: LogEntriesBaseRequest = {
        sourceId: overrides.sourceId ?? props.sourceId,
        startTimestamp: overrides.startTimestamp ?? props.startTimestamp,
        endTimestamp: overrides.endTimestamp ?? props.endTimestamp,
        query: overrides.filterQuery ?? props.filterQuery ?? undefined, // FIXME
      };

      const fetchArgs: LogEntriesRequest = props.timeKey
        ? {
            ...commonFetchArgs,
            center: props.timeKey,
          }
        : {
            ...commonFetchArgs,
            before: 'last',
          };

```

Reply:
I like it! I'll give it a try